deploy/examples/create-external-cluster-resources.py

```diff
@@ -693,6 +699,13 @@ def get_ceph_external_mon_data(self):
                 "Only 'v2' address type is enabled, user should also enable 'v1' type as well"
             )
         ip_port = str(q_leader_details["public_addr"].split("/")[0])
+
+        if self._arg_parser.v2_port_enable:
+            if q_leader_addrvec[0]["type"] == "v2":
```

Comment:
1st condition would never exist, Should check the length first

  if self._arg_parser.v2_port_enable and len(q_leader_addrvec) > 1 :
           then check q_leader_addrvec[1]["type"]
           or q_leader_addrvec[0]["type"]
  Something like this